src/simplewallet/simplewallet.cpp

```diff
@@ -8354,3 +8581,1025 @@ int main(int argc, char* argv[])
+  {
+    ms.add_message(state, i, mms::message_type::member_config, mms::message_direction::out, member_config);
+  }
+}
+
+void simple_wallet::show_message(const mms::message &m)
+{
+  mms::message_store& ms = m_wallet->get_message_store();
+  const mms::coalition_member &member = ms.get_member(m.member_index);
+  bool display_content;
+  switch (m.type)
+  {
+  case mms::message_type::key_set:
+  case mms::message_type::additional_key_set:
+  case mms::message_type::note:
+    display_content = true;
+    break;
+  default:
+    display_content = false;
+  }
+  uint64_t now = time(NULL);
+  success_msg_writer() << "";
+  success_msg_writer() << tr("Message ") << m.id;
+  success_msg_writer() << tr("In/out: ") << ms.message_direction_to_string(m.direction);
+  success_msg_writer() << tr("Type: ") << ms.message_type_to_string(m.type);
+  success_msg_writer() << tr("State: ") << boost::format(tr("%s since %s, %s ago")) %
+          ms.message_state_to_string(m.state) % get_human_readable_timestamp(m.modified) % get_human_readable_timespan(std::chrono::seconds(now - m.modified));
+  if (m.sent == 0)
+  {
+    success_msg_writer() << tr("Sent: Never");
+  }
+  else
+  {
+    success_msg_writer() << boost::format(tr("Sent: %s, %s ago")) %
+            get_human_readable_timestamp(m.sent) % get_human_readable_timespan(std::chrono::seconds(now - m.sent));
+  }
+  success_msg_writer() << tr("Member: ") << ms.member_to_string(member, 100);
+  success_msg_writer() << tr("Content size: ") << m.content.length() << tr(" bytes");
+  success_msg_writer() << tr("Content: ") << (display_content ? m.content : tr("(binary data)"));
+
+  if (m.type == mms::message_type::note)
+  {
+    // Showing a note and read its text is "processing" it: Set the state accordingly
+    // which will also delete it from Bitmessage as a side effect
+    // (Without this little "twist" it would never change the state, and never get deleted)
+    ms.set_message_processed_or_sent(m.id);
+  }
+}
+
+void simple_wallet::ask_send_all_ready_messages()
+{
+  mms::message_store& ms = m_wallet->get_message_store();
+  std::vector<mms::message> ready_messages;
+  const std::vector<mms::message> &messages = ms.get_all_messages();
+  for (size_t i = 0; i < messages.size(); ++i)
+  {
+    const mms::message &m = messages[i];
+    if (m.state == mms::message_state::ready_to_send)
+    {
+      ready_messages.push_back(m);
+    }
+  }
+  if (ready_messages.size() != 0)
+  {
+    list_mms_messages(ready_messages);
+    bool send = ms.get_auto_send();
+    if (!send)
+    {
+      send = user_confirms(tr("Send these messages now?"));
+    }
+    if (send)
+    {
+      mms::multisig_wallet_state state = get_multisig_wallet_state();
+      for (size_t i = 0; i < ready_messages.size(); ++i)
+      {
+        ms.send_message(state, ready_messages[i].id);
+        ms.set_message_processed_or_sent(ready_messages[i].id);
+      }
+      success_msg_writer() << tr("Sent.");
```

Comment:
That's indeed a weak point, and maybe a success message of "Sent." is overly optimistic and/or misleading.

If the MMS cannot contact PyBitmessage or if PyBitmessage returns an error code there will be an exception in `message_transporter` and a corresponding error message to the console instead of "Sent." If later however PyBitmessage has trouble to **really** send out the message over the Bitmessage network the CLI wallet won't be aware about any problem.

Reply:
With the exceptions thing, it's good enough.

Reply:
I changed the message to "Queued for sending."